analyzers/src/SonarAnalyzer.CSharp/Rules/BooleanLiteralUnnecessary.cs

```diff
@@ -99,5 +136,11 @@ private void CheckConditional(SonarSyntaxNodeReportingContext context)
 
         private static bool IsThrowExpression(ExpressionSyntax expressionSyntax) =>
             ThrowExpressionSyntaxWrapper.IsInstance(expressionSyntax);
+
+        private SyntaxNode GetIsPatternLeft(IsPatternExpressionSyntaxWrapper patternWrapper) =>
+            Language.Syntax.RemoveParentheses(patternWrapper.Expression);
+
+        private SyntaxNode GetIsPatternRight(IsPatternExpressionSyntaxWrapper patternWrapper) =>
+            Language.Syntax.RemoveParentheses(((ConstantPatternSyntaxWrapper)patternWrapper.Pattern).Expression);
```

Comment:
May need support for the `not` [UnaryPattern](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.syntax.unarypatternsyntax?view=roslyn-dotnet-4.3.0). You just need to dig one level deeper for this:
* IsPattern.Pattern { Kind: NotPattern } notPattern
* notPattern.Pattern is ConstantPattern { Expression: { } expression }